src/odemis/driver/xt_client.py

```diff
@@ -1621,14 +1644,10 @@ def _acquire(self):
                 while True:
                     if self._acq_should_stop():
                         break
-                    self._scanner.prepareForScan()  # In case of the FIB make sure the mode isn't in external
+                    # TODO When switching e-beam <--> FIB, handle calling finishScan on the old scanner and
+                    #  prepareForScan on the new scanner.
+                    self._scanner.prepareForScan()
                     self.parent.set_channel_state(self._scanner.channel, True)
-                    if hasattr(self._scanner, "blanker"):
-                        # TODO: When the acquisition ends, if blanker is set to automatic (None), we need to activate
-                        #  the blanker again. That should also happen when switching e-beam <--> FIB, but currently
-                        #  the FIB blanker is not supported via the XT interface.
-                        if self._scanner.blanker.value is None:
-                            self.parent.unblank_beam()
```

Comment:
The beam blanking is now done before the channel state is set to True. I tested with the simulator VM and this properly unblanks the beam.